### DISCUSSION TITLE
Could not build wheels for fastecdsa

### DISCUSSION THREAD

**Discussion by BalogunMalikO** [1200wd/bitcoinlib#306] [Q&A] [answered]
How can I use bitcoinlib on macbook? I get this error evrytime i try to pip install bitcoinlib

**Comment by AlexOQ** [comment]
The root cause is with `fastecdsa` package, the fix should come from them. But also, would be nice to have `bitcoinlib` version w/o this dependency (if possible), in case `fastecdsa` team takes their time.

EDIT:
OS: `Linux pop-os 6.2.6-76060206-generic #202303130630~1681329778~22.04~d824cd4`
Poetry: 1.4.2
Python: 3.10.6

**Comment by fled-dev** [comment]
Hi,
I had this error too. To fix that, simply run this command: `sudo apt-get install libgmp-dev`.

Helped for me at least. Let me know if you need any help / it does not work.

**Comment by fled-dev** [comment]
By the way, this is not an issue only with MacBooks. I also had this issue on my Ubuntu 23.04 server.

**Comment by mccwdev** [comment]
You need to install some required packages first before installing Bitcoinlib, see https://github.com/1200wd/bitcoinlib#pre-requirements-linux.

I did not run into any errors when installing on Ubuntu server 23.04

**Comment by superggn** [comment]
try use a virtual environment and change the cache dir

`python3 -m venv <the path you want to store the virtual environment>`

`pip install --cache-dir ~/<specify any temp cache dir> bitcoinlib`

works for me

**Comment by Juniorduc44** [comment, marked as answer]
## Safely work within a python environment

### First navigate to where you want to create the python environment on linux or Windows
    python3 -m venv venvLinux
###
    python -m venv venvWindows
### Second activate the environment on Linux or Windows
    source venvLinux/bin/activate
###
    venvWindows\Scripts\activate

## Upgrades & Installs
### Upgrade pip and install on Linux & Windows
    pip install --upgrade pip
###   
    pip install wheel

### Lastly, install fastecdsa
    python -m pip install fastecdsa

#### Deactivate the python environment with the following
    deactivate

**Reply by Juniorduc44** [reply]
After all that i was able to successfully pip install bitcoinlib
